tests/geo_test.go

```diff
@@ -0,0 +1,105 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/nknorg/tuna"
+)
+
+type testCase struct {
+	f        tuna.IPFilter
+	location tuna.Location
+	result   bool
+}
+
+var IP1 = "1.0.0.0"
+var IP2 = "2.0.0.0"
+var IP3 = "3.0.0.0"
+var IP4 = "4.0.0.0"
+
+var testData = []testCase{
+	{
+		f: tuna.IPFilter{
+			Allow: []tuna.Location{{IP: IP1}},
+		},
+		location: tuna.Location{IP: IP1},
+		result:   true,
+	},
+	{
+		f: tuna.IPFilter{
+			Disallow: []tuna.Location{{IP: IP1}},
+		},
+		location: tuna.Location{IP: IP1},
+		result:   false,
+	},
+	{
+		f: tuna.IPFilter{
+			Allow:    []tuna.Location{},
+			Disallow: []tuna.Location{},
+		},
+		location: tuna.Location{IP: IP1},
+		result:   true,
+	},
+	{
+		f: tuna.IPFilter{
+			Allow:    []tuna.Location{{IP: IP1}},
+			Disallow: []tuna.Location{{IP: IP1}},
+		},
+		location: tuna.Location{IP: IP1},
+		result:   true,
```

Comment:
Disallow should have priority over Allow and the expected result should be false